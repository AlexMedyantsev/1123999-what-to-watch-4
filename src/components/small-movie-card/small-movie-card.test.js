import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card.jsx";

const film = {
  image: `img/snatch.jpg`,
  title: `Snatch`,
};

describe(`it should SmallMovieCard render correctly`, () => {
  it(`Small card component`, () => {
    const tree = renderer
      .create(<SmallMovieCard
        backgroundImage={film.image}
        image={film.image}
        title={film.title}
        index={Math.random() * 100}
        onMovieCardClick={() => {}}
        onMovieCardHover={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
