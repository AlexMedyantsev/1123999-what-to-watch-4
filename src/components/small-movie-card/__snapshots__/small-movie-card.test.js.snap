// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should SmallMovieCard render correctly Small card component 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
>
  <div
    className="small-movie-card__image"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <video>
      Sorry, this preview is not available in your browser
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Name
    </a>
  </h3>
</article>
`;
